src/explore-education-statistics-common/src/modules/find-statistics/components/__tests__/DataBlockTabs.test.tsx

```diff
@@ -381,6 +383,78 @@ describe('DataBlockTabs', () => {
     });
   });
 
+  // EES-4902 react-leaflet now uses ESM so unable to test this currently
+  test('selecting data set with boundaryLevel retrieves and renders new map polygons', async () => {
+    tableBuilderService.getDataBlockTableData.mockResolvedValue(
+      testMapTableData,
+    );
+    tableBuilderService.getLocationGeoJson
+      .mockResolvedValueOnce(testMapTableData.subjectMeta.locations)
+      .mockResolvedValue(testMapTableDataLocationsLowRes);
+
+    const { container } = render(
+      <DataBlockTabs
+        releaseId="release-1"
+        id="test-block"
+        dataBlock={testDataBlockMap}
+      />,
+    );
+
+    forceVisible();
+
+    await waitFor(() => {
+      expect(tableBuilderService.getDataBlockTableData).toBeCalledWith(
+        'release-1',
+        'block-1-parent',
+      );
+
+      expect(tableBuilderService.getLocationGeoJson).toBeCalledWith(
+        'release-1',
+        'block-1-parent',
+        1,
+      );
+
+      // expect(container.querySelector('.leaflet-container')).toBeInTheDocument();
+    });
+    expect(tableBuilderService.getDataBlockTableData).toHaveBeenCalledTimes(1);
+
+    await waitFor(() => {
+      expect(tableBuilderService.getLocationGeoJson).toBeCalled();
+      expect(tableBuilderService.getLocationGeoJson).toBeCalledTimes(1);
+    });
+
+    userEvent.click(screen.getByRole('link', { name: 'Chart for' }));
+
+    await waitFor(() => {
+      expect(
+        screen.getByLabelText('1. Select data to view'),
+      ).toBeInTheDocument();
+    });
+
+    // TODO: check for path/polygon node, save position(s) to compare later (EES-4902)
+
+    const dataSetSelectInput = screen.getByLabelText('1. Select data to view');
+    const dataSetOptions = within(dataSetSelectInput).getAllByRole('option');
```

Comment:
As other places - could simplify these variable names to `select` and `options` respectively